lib/fizzy/parser_expr.cpp

```diff
@@ -91,6 +93,10 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, bool have
 
         frame.stack_height += metrics.stack_height_change;
 
+        if (!frame.unreachable)
+            func_stack_height += metrics.stack_height_change;
```

Comment:
What about `block`, `loop`, `if_`? I think non-0-arity block should affect `func_stack_height`, too?

Reply:
No, they don't (but please validate that claim, because I was thinking the same as you at first).
They affect validation. E.g.  for `(block (result i32))` the validation must make sure that there is exactly one item on the stack when reaching `end` of this block. So here we are safe to only follow every other instruction stack change.

Reply:
Yeah I see, sounds right. Block start/end doesn't pop/push anything by itself, only the instructions inside do.